extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -345,6 +349,163 @@ public void startPersistenceUnits(HibernateOrmRecorder recorder, BeanContainerBu
         recorder.startAllPersistenceUnits(beanContainer.getValue());
     }
 
+    @BuildStep
+    public void metrics(HibernateOrmConfig config,
+            BuildProducer<MetricBuildItem> metrics) {
+        // TODO: When multiple PUs are supported, create metrics for each PU. For now we only assume the "default" PU.
+        boolean metricsEnabled = config.metricsEnabled && config.statistics.orElse(true);
+        metrics.produce(createMetricBuildItem("hibernate.sessions.open",
+                "Global number of sessions opened",
+                "sessionsOpened",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.sessions.closed",
+                "Global number of sessions closed",
+                "sessionsClosed",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.sessions.closed",
+                "Global number of sessions closed",
+                "sessionsClosed",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.transactions",
+                "The number of transactions we know to have completed",
+                "transactionCount",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.transactions.successful",
+                "The number of transactions we know to have been successful",
+                "successfulTransactions",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.optimistic.lock.failures",
+                "The number of Hibernate StaleObjectStateExceptions or JPA OptimisticLockExceptions that occurred.",
+                "optimisticLockFailures",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.flushes",
+                "Global number of flush operations executed (either manual or automatic).",
+                "flushes",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.connections.obtained",
+                "Get the global number of connections asked by the sessions " +
+                        "(the actual number of connections used may be much smaller depending " +
+                        "whether you use a connection pool or not)",
+                "connectionsObtained",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.statements.prepared",
+                "The number of prepared statements that were acquired",
+                "statementsPrepared",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.statements.closed",
+                "The number of prepared statements that were released",
+                "statementsClosed",
+                metricsEnabled));
+        metrics.produce(createMetricBuildItem("hibernate.second.level.cache.puts",
```

Comment:
Is it the norm to use dots everywhere? `hibernate.second-level-cache.puts` would have felt more natural to me.

Reply:
Ok. I changed `natural.id` and `query.cache` and `timestamps.cache` too in a similar fashion